paimon-core/src/main/java/org/apache/paimon/iceberg/manifest/IcebergConversions.java

```diff
@@ -74,8 +77,34 @@ public static ByteBuffer toByteBuffer(DataType type, Object value) {
             case DECIMAL:
                 Decimal decimal = (Decimal) value;
                 return ByteBuffer.wrap((decimal.toUnscaledBytes()));
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                TimestampType timestampType = (TimestampType) type;
+                return convertTimestampWithPrecisionToBuffer(
+                        (Timestamp) value, timestampType.getPrecision());
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                LocalZonedTimestampType localTimestampType = (LocalZonedTimestampType) type;
+                return convertTimestampWithPrecisionToBuffer(
+                        (Timestamp) value, localTimestampType.getPrecision());
+            case TIME_WITHOUT_TIME_ZONE:
+                long microsecondsFromMidnight = (Integer) value / 1_000;
```

Comment:
Why casting it to integer? It is a `long` value. One second is $10^9$ nano seconds and the maximum value of `int` is about $2 \times 10^9$. You won't expect it to overflow after only three seconds from midnight.